File: LeetCode/0509. Fibonacci Number/solution.py
"""
31 / 31 test cases passed.
Status: Accepted
Runtime: 44 ms
"""
class Solution:
    def fib(self, N: int) -> int:
        a, b, ans = 0, 1, 0
        for i in range(N):
            ans = b
            b = a + b
            a = ans
        return ans

File: LeetCode/0509. Fibonacci Number/solution.cpp
/**
 * 31 / 31 test cases passed
 * Status: Accepted
 * Runtime: 0 ms
 */
class Solution {
public:
    int fib(int N) {
        int a = 0, b = 1, ans = 0;
        for (int i = 1; i <= N; i++) {
            ans = b;
            b = a + b;
            a = ans;
        }
        return ans;
    }
};
